doc/man1/openssl.pod

```diff
@@ -778,11 +778,266 @@ client.
 
 =item B<-xcertform> B<DER>|B<PEM>, B<-xkeyform> B<DER>|B<PEM>
 
+The input format for the extra certifcate and key, respectively.
+See L<openssl(1)/Format Options> for details.
+<<<<<<< dc5af7967d7a4b6afd777a460f0c8918768e256e
+=======
+
+=item B<-xchain_build>
+
+Specify whether the application should build the certificate chain to be
+provided to the server for the extra certificates via the B<-xkey>,
+B<-xcert>, and B<-xchain> options.
+>>>>>>> Move verify options to openssl.pod
+
+=item B<-xcertform> B<DER>|B<PEM>, B<-xkeyform> B<DER>|B<PEM>
+
 The input format for the extra certifcate and key, respectively.
 See L<openssl(1)/Format Options> for details.
 
 =back
 
+=head2 Verification Options
+
+Many OpenSSL commands verify certificates.  The commands try to proceed
+after any validation error, in order to allow all the problems with a
+certificate chain to be determined.
+
+Verification is a complicated process, consisting of a number of separate
+steps that are detailed in the following paragraphs.
+
+First, a certificate chain is built up starting from the supplied certificate
+and ending in a root CA.  It is an error if the whole chain cannot be
+built up.  The chain is built up by looking up the issuers certificate
+of the current certificate.  If a certificate is found which is its own
+issuer it is assumed to be the root CA.
+
+The process of looking up the issuer's certificate itself involves a number
+of steps.  After all certificates whose subject name matches the issuer
+name of the current certificate are subject to further tests.  The relevant
+authority key identifier components of the current certificate (if present)
+must match the subject key identifier (if present) and issuer and serial
+number of the candidate issuer, in addition the keyUsage extension of the
+candidate issuer (if present) must permit certificate signing.
+
+The lookup first looks in the list of untrusted certificates and if no match
+is found the remaining lookups are from the trusted certificates. The root CA
+is always looked up in the trusted certificate list: if the certificate to
+verify is a root certificate then an exact match must be found in the trusted
+list.
```

Comment:
Here, you'll need to reconcile with #10587 or perhaps 10587 refactored to follow your patch...

Reply:
Yes, that's very very frustrating.  This PR has sat her since *OCTOBER 9* and nobody has looked at it until now.  I have rebased, posted comments, and so on.

If #10587 goes in before this, then I hope someone else will provide a patch because I doubt that I will do it. There is a cost to ignoring PR's and this is a prime example. By letting this languish, the project has forced someone (@ddvo or me), who is only trying to help improve documentation and code, to do needless extra work. Maybe the project will try to be more diligent in the future so that this type of problem can be avoided.

Reply:
I think your PR should be merged first, and then #10587. I hope that's OK.
Not sure how important the fine distinctions between s_client/s_server and the other apps (verify, ...) are to document, but lean towards documenting the actual behaviour.